### DISCUSSION TITLE
Windows Terminal v1.17.11391.0

### DISCUSSION THREAD

**Discussion by DHowett** [microsoft/terminal#15405] [Announcements]
Windows Terminal 1.17 is finally out of preview! It includes...

* The ability to customize the order and contents of the New Tab menu!
* `scrollbarState` `always`, for those of you who want to have a scrollbar _constantly_.
* Last-chance profile restarting with <kbd>Enter</kbd> after it exits! Excellent for those IoT devices you SSH into!

Please see the following release notes for additional details:

- [Windows Terminal Preview v1.17.1023](https://github.com/microsoft/terminal/releases/tag/v1.17.1023)

> **Note** The change from [v1.17.1073](https://github.com/microsoft/terminal/releases/v1.17.10230) where we would munge our version number didn't live much longer than a single release. We finally rejoined the Windows 10 and Windows 11 packages!

Note that the new scrollbar mark feature is still not included in this Stable build (?)

### Why are there so many packages? How do I choose?

Please visit our page documenting the [Windows Terminal Distributions]!

## Features

* We have made some changes to the officially supported list of Terminal distributions!
   * The "_unpackaged_" distribution of Terminal is now officially supported on Windows 10 (10.0.19041.1 and higher) and
     Windows 11 (10.0.22000.1 and higher)! (#15034)
   * We now offer support for running in [portable mode], where settings are saved into a directory alongside the
     app (#15051)
   * If you have multiple unpackaged or portable installations of Terminal, they will now properly isolate from
     eachother (#15118)
   * **LEARN MORE** at our new documentation page for [Windows Terminal Distributions]!
   * In 1.18 and above: We've removed any runtime dependency on the "desktop" C++ Runtime Library (#14733) (#15010) (#15097)
      * If you are installing 1.17 or below, you still need to install a VC++ runtime. Even in portable mode. Sorry.
* We have added another way of detecting whether "Default Terminal" is available on your OS :eyes: (#15040)

## Changes

* `file:` URIs emitted by `OSC 8` now support `wsl$`; URIs emitted by `OSC 8` now support all valid schemas (#14993)
* The "Adjust Indistinguishable Text" feature now uses the Oklab color space (by Björn Ottosson) (#15283)
   * It no longer has a measurable performance impact, either, so feel free to turn it on!
* Any WSL distributions named `rancher-desktop` or `rancher-desktop-data` will now be ignored (#15166) (thanks @jamespack!)
   * Really sorry if you were using those for your day-to-day WSLing, but... _what?_

### User Interface and Usability

* The Profile Appearance preview now displays a more relevant and interesting preview (#15363)
* We rewrote a bunch of the subtitles in the Settings UI for diction and flow (#15362) (#15361)
* Rejoice! We have reintroduced the original Windows Terminal UI themes under the names `Dark (Legacy)`, `Light (Legacy)` and `System (Legacy)` (#15108)
* You will now be reprimanded ~~harshly~~ for choosing a proportional font (#15195) (thanks @jamespack!)

### Performance

* The "Azure Cloud Shell" connection no longer depends on "cpprestsdk", the removal of which shrinks our package
  by about 3MB per architecture! (#14776)
   * Due to differences in credential storage, this will require you to go through the Device Code authentication flow
     again.
* Terminal no longer needs to shell out to `reg.exe` to manipulate the registry to work around a platform limitation
  (#15028) (#15050)

## Bug Fixes

* Starting Terminal with an `elevate:true` profile _should_ no longer cause an admin elevation loop (oops) (#14946)
* "Open in Terminal" no longer shows up in ZIP files (???) (#15236) (thanks @jamespack!)
* The "Developer Command Prompt / PowerShell" profiles no longer override your starting directory (#15035)
* Copying HTML to your clipboard no longer results in reeeeeally tiny text on high-DPI displays (#15046)
* On very wide monitors, the `Outlines` shader sample no longer looks quite bad (#14971)

### Accessibility

* The about dialog heading is now read out properly to a screen reader (#15200) (thanks @BenConstable9!)
* The "reset color" button in the tab color picker is now more visible to screen readers (#15257) (thanks @BenConstable9!)

### UI and stuff like it

* @jamespack has added epxlanatory text to "Always Show Tabs" explaining why it can't be turned off sometimes (#15154) (thanks @jamespack!)
* Opening a profile from a submenu no longer yeets keyboard focus off into space (#15077) (thanks @kkostrzewa>!)
* When you're hovering over a URL, we now display it in a partially-encoded form to help you avoid homoglyph attacks (#15095)

### Reliability and Performance

* Animations are disabled during startup so as to make Terminal start a bit faster (#15204)
* The system menu now opens when you press `<kbd>Alt+Space</kbd>` in the Settings UI (#15189) (thanks @jamespack!) with bugfixes (#15253)

### VT and Compatibility

* We'll no longer crash when VTParameters::subspan is out of range (#15235) (thanks @j4james!)

## Code Health, WPF and Windows Console

_These changes impact the Terminal developer experience or the Windows Console Host. Changes to conhost will be released
in a coming Windows update._

* We have removed all traces of the UWP Terminal (which was an internal build for Windows 10X) (#15222)
* The WPF control now defaults to a history length of 9001 (#15129)

---

### Asset Hashes

- Microsoft.WindowsTerminal_1.17.11391.0_8wekyb3d8bbwe.msixbundle
   - SHA256 `2C46A63C764C37DD602250F9173E30603F708780CB3D55C7381EE8126311B828`
- Microsoft.WindowsTerminal_1.17.11391.0_8wekyb3d8bbwe_Windows10_PreinstallKit.zip
   - SHA256 `BB943E0CAA7569993DB906C7A9B386E122B95DC1074BD47BE363802450DD1DF6`
- Microsoft.WindowsTerminal_1.17.11391.0_arm64.zip
   - SHA256 `F483D0D77097402DBCB3F40C60124A735D8E5FF30F7F1E2DA5412A17E0666A6A`
- Microsoft.WindowsTerminal_1.17.11391.0_x64.zip
   - SHA256 `061E019C311592212C4A8C36ABDBDACACB2D875DE055FB3E28602900A35EE617`
- Microsoft.WindowsTerminal_1.17.11391.0_x86.zip
   - SHA256 `5BFD0E7E0EC3BBC54201836F316AC6484E0EC9D192A77C1E24A34C4EC95C28CB`

[portable mode]: https://go.microsoft.com/fwlink/?linkid=2229086
[Windows Terminal Distributions]: https://learn.microsoft.com/windows/terminal/distributions

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/terminal/releases/tag/v1.17.11391.0'>Windows Terminal v1.17.11391.0</a>.</em>

**Comment by ghost** [comment]
Portable mode is so useful. Thank you.

**Reply by DHowett** [reply]
Thanks for letting us know! I'm really glad it's helped someone :)
